File: Src/ButterflyNet.Parser/Scopes/DefinitionScope.cs
namespace ButterflyNet.Parser.Scopes {
	public class DefinitionScope : BlockScope {
		public override void Open(ButterflyAnalyzer analyzer) {
			analyzer.OpenDefinition();
		}

		public override void Close(ButterflyAnalyzer analyzer) {
			analyzer.CloseDefinition();
		}
	}
}

File: src/js/src/scopes/DefinitionScope.js
function DefinitionScope() {
	this.type = ScopeTypeCache.definition;
	this.level = ScopeLevel.block;
	this.canNestText = true;
	
	this.open = function(analyzer) {
		analyzer.openDefinition();
	};
	this.close = function(analyzer) {
		analyzer.closeDefinition();
	};
}

extend(Scope, DefinitionScope);